canopeum_frontend/src/constants/weatherImageMap.ts

```diff
@@ -62,9 +59,11 @@ const getImageNameByWMOCategories = (WMOCategory: string): string | undefined =>
       wmoCategoriesImage.WMOCategories.some(category => WMOCategory.includes(category))
     ) ?? undefined
 
-  return image
-    ? image.imageUrl
-    : WMOCategoriesImages[0].imageUrl
+  return image && image.imageUrl
+    ? image.imageUrl.href
+    // image is defined manually, so it is guaranteed to have an imageUrl
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- comment above
+    : WMOCategoriesImages[0].imageUrl!.href
```

Comment:
The return type is already possibly undefined

```suggestion
    : WMOCategoriesImages[0].imageUrl?.href
```